tests/unit/plugins/test_enricher.py

```diff
@@ -19,18 +19,132 @@
+
+
+@pytest.fixture
+def mock_http_client(mocker, get_mock_coro):
+    get_json_mock, get_json_coro = get_mock_coro()
+    http_client = mocker.Mock()
+    mocker.patch.object(http_client, 'get_json', get_json_coro)
+    mocker.patch.object(http_client, '_get_json', get_json_mock)
+    return http_client
+
+
+@pytest.fixture
+def mock_async_sleep(mocker, get_mock_coro):
+    sleep_mock, sleep_coro = get_mock_coro()
+    mocker.patch('gordon_gcp.plugins.enricher.asyncio.sleep', sleep_coro)
+    return sleep_mock
+
+
+@pytest.mark.parametrize('network_data,sleep_calls,logs_logged', [
+    ([INSTANCE_NETWORK_DATA], 0, 3),
+    ([{}, {}, INSTANCE_NETWORK_DATA], 2, 5)])
+@pytest.mark.asyncio
+async def test_process_event_msg(config, event_msg, channel_pair,
+                                 mock_async_sleep, mock_http_client, caplog,
+                                 network_data, sleep_calls, logs_logged):
+    success_chnl, error_chnl = channel_pair
+    mock_http_client._get_json.side_effect = network_data
+    gce_enricher = enricher.GCEEnricher(config, mock_http_client, success_chnl,
+                                        error_chnl)
+
+    await gce_enricher.process(event_msg)
+
+    event_message = await success_chnl.get()
+
+    assert 'publish' == event_message.phase
+    expected_history_msg = 'Enriched msg 1 with resourceRecords'
+    assert expected_history_msg == event_message.history_log[0]['message']
+    assert 1 == len(event_message.data['resourceRecords'])
+    expected_rrecords = [{
+        'name': '.'.join([
+            event_msg.data['resourceName'].split('/')[-1], config['dns_zone']
+        ]),
+        'type': 'A',
+        'rrdatas': [
+            INSTANCE_NETWORK_DATA['networkInterfaces'][0]['accessConfigs'][0][
+                'natIP']
+        ]
+    }]
+    assert expected_rrecords == event_message.data['resourceRecords']
+    assert sleep_calls == mock_async_sleep.call_count
+    assert logs_logged == len(caplog.records)
+
+
+@pytest.mark.parametrize('response,sleep_calls,logs_logged', [
+    (exceptions.GCPHTTPError('404 error'), 0, 2),
+    ([{}] * 5, 4, 6)])
+@pytest.mark.asyncio
+async def test_process_event_msg_failures(config, event_msg, channel_pair,
+                                          mock_async_sleep, mock_http_client,
+                                          caplog, response, sleep_calls,
+                                          logs_logged):
+    success_chnl, error_chnl = channel_pair
+    mock_http_client._get_json.side_effect = response
+    gce_enricher = enricher.GCEEnricher(config, mock_http_client, success_chnl,
+                                        error_chnl)
+
+    await gce_enricher.process(event_msg)
+
+    event_message = await error_chnl.get()
+
+    assert 'enrich' == event_message.phase
+    assert event_msg == event_message
+    assert sleep_calls == mock_async_sleep.call_count
+    assert logs_logged == len(caplog.records)
+    assert 0 == len(event_message.history_log)
```

Comment:
Hmm maybe there should be a log entry for the message history about this. What do you think?